src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionPool.cs

```diff
@@ -351,6 +365,33 @@ public bool CleanCacheAndDisposeIfUnused()
             return false;
         }
 
+        public override string ToString() => $"{nameof(HttpConnectionPool)}(Connections:{_associatedConnectionCount})"; // Description for diagnostic purposes
+
+        private void Trace(string message, [CallerMemberName] string callerName = null)
```

Comment:
I thought the general pattern here was to put the IsEnabled at the call site, to avoid stuff like evaluating parameters.